### PR TITLE
`ASAP-Alchemy`: BespokeFit integration

### PR DESCRIPTION
**Pull request by jthorton**
## Description
This PR extends the `ASAP-Alchemy` CLI to interface with `openff-bespokefit`. This exposes three new commands 
- `asap-alchemy bespoke submit`: Submit the ligands in a local network to a bespokefit server
- `asap-alchemy bespoke status`: Get the status breakdown for ligands in a local network
- `asap-alchemy bespoke gather`: Get the bespoke parameters derived for the ligands in the local network

We have also updated the `asap-alchemy submit` command to automatically inject the bespoke parameters required for a given transformation into the force field when creating the OpenFE inputs. Note that only the parameters needed for the edge are included which should trigger the deduplication of the task in alchemiscale as more bespoke parameters are derived for a chemical series. 

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [x] tests
- [x] update cli docs
- [ ] update tutorial docs 

## Questions
- [ ] Question 1 ..

## Status
- [x] Ready to go

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by jthorton** [comment]
Testing is blocked, till we get a new bespokefit release and an alchemiscale release which works with gufe and openfe >=1.

**Comment by hmacdope** [comment]
With @jthorton we are pursuing a different path where we just pretrain a forcefield and host as an artifact. 

Due to the way bespokefit does its server-worker communication, there is a need to open  database http ports on the open internet which is not a good idea. 

We will leave this open as an example of how this might be done for the future.

**Comment by jthorton** [comment]
@hmacdope & @JenkeScheen  can I get a quick look over this? We need this to submit the calculations for the bespokefit benchmarking we can run it from the branch though if we don't want this in main.

File: asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py [hunk 1]
```diff
@@ -20,31 +20,15 @@
...
 
 from ._util import check_ligand_series_uniqueness_and_names
 from .base import _SchemaBase, _SchemaBaseFrozen
 from .network import NetworkPlanner, PlannedNetwork
 
 if TYPE_CHECKING:
     from asapdiscovery.data.schema.ligand import Ligand
-
-
-class AlchemiscaleSettings(BaseSettings):
```
Comment (hmacdope) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py:33]: Good idea to move.

File: asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py [hunk 2]
```diff
@@ -363,13 +359,87 @@ def to_alchemical_network(self) -> openfe.AlchemicalNetwork:
...
+                        # similar ligands will share parameters so make sure we don't add it twice
+                        handler.add_parameter(parameter_kwargs=parameter_data)
+                    except DuplicateParameterError:
+                        continue
+
+        # if we found bespoke parameters return the new force field
+        if bespoke_parameters:
+            ff_string = ff.to_string()
+
+        return ff_string
```
Comment (hmacdope) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py:441]: Very nice.

File: asapdiscovery-data/asapdiscovery/data/schema/ligand.py [hunk 3]
```diff
@@ -124,6 +125,11 @@ class Ligand(DataModelAbstractBase):
         None, description="The provenance information of the local charging method."
     )
 
+    bespoke_parameters: Optional[BespokeParameters] = Field(
```
Comment (hmacdope) [inline asapdiscovery-data/asapdiscovery/data/schema/ligand.py:128]: Nice.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Looking fantastic, my only request is that you write a Notion or docs document showing how this can be used in practice including with the bespokefit server on HPC. :) Can either go on ASAP notion or in the docs.

**Comment by jthorton** [comment]
Thanks @hmacdope added some simple CLI docs on how to use the interface let me know what you think!

**Comment by jthorton** [comment]
@hmacdope just noticed the docs are not building on the PR any more not sure what has happened

**Review by hmacdope — APPROVED** [review]

**Pull request merged**